comm/src/main/scala/coop/rchain/comm/transport/StreamHandler.scala

```diff
@@ -120,10 +120,10 @@ object StreamHandler {
           stmd.fos.write(array)
           stmd.fos.flush()
           val readSoFar = stmd.readSoFar + array.length
-          if (circuitBreaker(readSoFar))
-            Right(stmd.copy(circuitBroken = true))
-          else
-            Left(stmd.copy(readSoFar = readSoFar))
+          val newStmd   = stmd.copy(readSoFar = readSoFar)
+          if (circuitBreaker(newStmd))
+            Right(newStmd.copy(circuitBroken = true))
```

Comment:
because if it was `opened/broken` there was no point, but this is correct i guess: we open circuit because we've read to much data. number of data read is not passed to error message, maybe it should?